.github/workflows/ci.yaml

```diff
@@ -964,23 +964,35 @@ jobs:
         run: |
           ./script/check_dirty
 
-  coverage:
-    name: Upload test coverage to Codecov
+  coverage-partial:
+    name: Upload partial test coverage to Codecov
+    runs-on: ubuntu-20.04
+    needs:
+      - info
+      - pytest
+    if: needs.info.outputs.test_full_suite != 'true'
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v3.1.0
+      - name: Download all coverage artifacts
+        uses: actions/download-artifact@v3
+      - name: Upload coverage to Codecov (partial coverage)
+        uses: codecov/codecov-action@v3.1.1
+
+  coverage-full:
+    name: Upload full test coverage to Codecov
     runs-on: ubuntu-20.04
     needs:
       - info
       - pytest
       - pytest-mariadb
```

Comment:
I think, alternatively, it is safe to just remove `pytest-mariadb` completely.
Although, this job won't wait, it won't harm either.

In almost all cases, the mariadb job will finish way before the other test jobs. Worst case: it misses a tiny bit of coverage at time. We'll notice that when it happens.